src/ZoneServer/Scripting/AI/AiScript.Routines.cs

```diff
@@ -98,7 +98,10 @@ protected IEnumerable Say(string message)
 		/// <exception cref="ArgumentException"></exception>
 		protected IEnumerable Emoticon(string packetString)
 		{
-			Send.ZC_SHOW_EMOTICON(this.Entity, packetString, TimeSpan.FromSeconds(2));
+			if (!ZoneServer.Instance.Data.PacketStringDb.TryFind(packetString, out var data))
```

Comment:
It seems like you reverted quite a few of my changes, so I'll once more postpone looking over the rest of this PR^^" Please check the comments I made and try not to revert recent commits when you merge.

Reply:
I am sorry about it, I am trying to save this PR somehow.